pkg/datastore/database.go

```diff
@@ -56,10 +57,12 @@ const (
 Postgres-backed implementation of DataStore interface. By default, uses MetadataBasedAuthorizer for authentication & authorization.
 */
 type relationalDb struct {
-	authorizer  authorizer.Authorizer // Allows or cancels operations on the DB depending on user's org. and service roles
-	gormDBMap   map[dbrole.DbRole]*gorm.DB
-	logger      *logrus.Entry
-	initializer func(db *relationalDb, dbRole dbrole.DbRole) error
+	sync.RWMutex
+	authorizer         authorizer.Authorizer // Allows or cancels operations on the DB depending on user's org. and service roles
+	gormDBMap          map[dbrole.DbRole]*gorm.DB
+	logger             *logrus.Entry
+	initializer        func(db *relationalDb, dbRole dbrole.DbRole) error
+	triggerFuncCreated bool
```

Comment:
I propose that the `check_and_update_revision` trigger function is created by default within `initializer`.  If no struct registered with DAL needs revisioning, no big deal, there will just be one extra function in the DB. We'll be able to get rid of `sync.RWMutex` and `triggerFuncCreated` if the trigger function is created in `initializer`.